packages/experimental/Menu/src/MenuItemRadio/MenuItemRadio.tsx

```diff
@@ -0,0 +1,30 @@
+/** @jsx withSlots */
+import { mergeProps, UseSlots, withSlots } from '@fluentui-react-native/framework';
+import { useMenuItemRadio } from './useMenuItemRadio';
+import { MenuItemCheckbox } from '../MenuItemCheckbox/MenuItemCheckbox';
+import { MenuItemCheckboxProps, MenuItemCheckboxType } from '../MenuItemCheckbox/MenuItemCheckbox.types';
+
+export const menuItemRadioName = 'MenuItemRadio';
+
+export const MenuItemRadio = MenuItemCheckbox.compose({
+  displayName: menuItemRadioName,
+  useRender: (userProps: MenuItemCheckboxProps, useSlots: UseSlots<MenuItemCheckboxType>) => {
+    const menuItem = useMenuItemRadio(userProps);
+    const Slots = useSlots(userProps, (layer): boolean => menuItem.state[layer]);
+
+    return (final: MenuItemCheckboxProps) => {
+      const mergedProps = mergeProps(menuItem.props, final);
+      const chevronXml = `
+      <svg>
+        <path fill='currentColor' d='M9.85355 3.14645C10.0488 3.34171 10.0488 3.65829 9.85355 3.85355L5.35355 8.35355C5.15829 8.54882 4.84171 8.54882 4.64645 8.35355L2.64645 6.35355C2.45118 6.15829 2.45118 5.84171 2.64645 5.64645C2.84171 5.45118 3.15829 5.45118 3.35355 5.64645L5 7.29289L9.14645 3.14645C9.34171 2.95118 9.65829 2.95118 9.85355 3.14645Z' />
+      </svg>`;
+
+      return (
+        <Slots.root {...mergedProps}>
+          <Slots.checkmark xml={chevronXml} />
+          {mergedProps.content && <Slots.content>{mergedProps.content}</Slots.content>}
+        </Slots.root>
+      );
+    };
+  },
```

Comment:
What's different between the `useRenders` of MenuItemRadio and MenuItemCheckbox? Or is it that using `.compose` means you have to specify it twice?

Reply:
useMenuItemRadio vs. useMenuItemCheckbox is the only difference.

But yeah, I have to re-specify the useRender again if I want to change it. I could refactor this too and maybe pass in a hook to run for state?

Reply:
I feel like to do that, you would need to pass the resulting `useMenuItemCheckbox` hook an extra boolean or prop that does not match the final component API. AKA, the props passed into `MenuItemRadio / MenuItemCheckbox` would not match the props passed into `useMenuItemCheckbox`, since we need at least 1 extra one to specify if it's a radio or not. Is that OK? I feel like the intention of these state hooks we have is to have their input be equal to the final components input?

Reply:
>  I feel like the intention of these state hooks we have is to have their input be equal to the final components input?

Dunno if I've ever heard of that. Is that a general pattern with state hooks?

Reply:
If it's not, then I guess it's fine

Reply:
I don't think it's an explicit rule, more of a general pattern? Anyway, I think I can work around it.

Reply:
I tried some refactoring, is it better? Or should I revert the .tsx files?

Reply:
(Post merge comment) I think it's better? Hard to say without locally cloning the PR myself, since both versions are pretty involved changes =) . I'll leave that as a problem for the next time I have to touch the code :D